kpconv_torch/io/las.py

```diff
@@ -0,0 +1,78 @@
+import laspy
+import numpy as np
+
+
+def read_las_laz(filepath):
+    """Takes a file path pointing on a 3D point .las or .laz file and returns the points,
+    the associated colors and the associated classes.
+
+    Parameters
+    ----------
+    filepath: path to a 3D points file with .laz or .las format
+
+    Colors are original encoded on 2 bytes
+    (cf. https://www.asprs.org/a/society/committees/standards/asprs_las_spec_v13.pdf), that is on a uint16.
+    They are converted (divided by 256) when the las or the laz file is read by this function.
+
+    Returns
+    -------
+    points: 2D np.array with type float32
+    colors: 2D np.array with type uint8
+    labels: 1D np.array with type int32
+    """
+    data = laspy.read(filepath)
+    points = np.vstack([data.x, data.y, data.z]).transpose().astype(np.float32)
+    colors = (
+        np.vstack([data.red / 256, data.green / 256, data.blue / 256]).transpose().astype(np.uint8)
+    )
+    labels = np.array(data.classification).astype(np.int32)
+
+    return points, colors, labels
+
+
+def write_las(filepath, points, colors, labels):
+    """Creates a .las file from a 3D point cloud.
+    Parameters
+    ----------
+    filepath: path to the .las file
+    points: 2D np.array with type float32
+    colors: 2D np.array with type uint8 or uint16
+    labels: 1D np.array with type int32
+
+    Uses a point_format = 6 (cf. https://pythonhosted.org/laspy/tut_background.html)
+    Warning: laspy does not have support for writing files in the .laz format.
+    """
+    hdr = laspy.LasHeader(point_format=7)
+    hdr.offsets = np.min(points, axis=0)
+    hdr.scale = [0.001, 0.001, 0.001]
+    las = laspy.create(point_format=7)
+    if points.dtype == np.dtype("float32"):
+        las.x = points[:, 0]
+        las.y = points[:, 1]
+        las.z = points[:, 2]
+    else:
+        raise TypeError(
+            f"Error while writing file {filepath}: points should be a float32 array, not {colors.dtype}"
+        )
+    if colors.dtype == np.dtype("uint8") or colors.dtype == np.dtype("int8"):
```

Comment:
```suggestion
    if colors.dtype == np.dtype(np.uint8) or colors.dtype == np.dtype(np.int8):
```
(Be consistent with the type declaration: always use `numpy` API.)